tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -308,6 +308,13 @@ TEST_F(OverlapTest, ReduceScatterBasedPipeliningHostIrImplementation) {
   for (Expr* expr : loop_body) {
     for_loop->body().push_back(expr);
   }
+  if (params.use_different_streams) {
+    for_loop->body().insert(
+        /*pos=*/0,
+        IrBuilder::create<hir::SetCurrentStream>(
+            IrBuilder::create<hir::Stream>(j)));
```

Comment:
I'd add a test to exercise that the same integer maps to the same CUDA stream. This test doesn't do that IIUC. Removing `j` would have the same effect -- creating a new stream each time the program enters the loop body.

Reply:
> I'd add a test to exercise that the same integer maps to the same CUDA stream. This test doesn't do that IIUC. 

Correct, I can add a test for that, Let me just clarify that the overlap technic doesn't need this feature.

> Removing `j` would have the same effect -- creating a new stream each time the program enters the loop body.

No, only one stream would be created, on the first iteration, corresponding to the one and only `Stream*` present in the fusion. Then, on each iteration, the same cuda stream would be set, over and over again.

Reply:
> Then, on each iteration, the same cuda stream would be set, over and over again.

Yes, that's right -- I missed that.